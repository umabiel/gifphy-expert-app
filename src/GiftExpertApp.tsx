import React, { Fragment, useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

interface IProps {
  defaultCategories: string[];
}

export const GiftExpertApp = ({ defaultCategories = [] }: IProps) => {
  const [categories, setCategories] = useState(defaultCategories);
  // "Samurai X",
  // "Dragon Ball",

  //   const handleAdd = () => {
  //     setCategories([...categories, "HunterXHunter"]);
  //   };

  return (
    <Fragment>
      <h2>GiftExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ol>
        {categories.map((cat) => {
          return <GifGrid category={cat} key={cat} />;
        })}
      </ol>
    </Fragment>
  );
};
