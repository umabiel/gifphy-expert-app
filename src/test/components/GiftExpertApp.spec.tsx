import React from "react";
import { shallow } from "enzyme";
import { GiftExpertApp } from "../../GiftExpertApp";

describe("Pruebas en <GiftExpertApp />", () => {
  test("Debe renderizar correctamente <GiftExpertApp />", () => {
    const categories = [] as string[];
    const wrapper = shallow(<GiftExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar una lista de <GifGrid />", () => {
    const categories = ["One Punch", "Dragon Ball"];
    const wrapper = shallow(<GiftExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toEqual(categories.length);
  });
});
