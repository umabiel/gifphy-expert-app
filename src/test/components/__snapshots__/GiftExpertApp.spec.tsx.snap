// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftExpertApp /> Debe de mostrar una lista de <GifGrid /> 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GiftExpertApp /> Debe renderizar correctamente <GiftExpertApp /> 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
